arch/arm/core/aarch32/cpu_idle.S

```diff
@@ -62,19 +62,36 @@ SECTION_FUNC(TEXT, arch_cpu_idle)
 #endif /* CONFIG_ARMV6_M_ARMV8_M_BASELINE */
 #endif /* CONFIG_TRACING */
 
-#if defined(CONFIG_ARMV6_M_ARMV8_M_BASELINE) \
-	|| defined(CONFIG_ARMV7_R)
-	cpsie i
-#elif defined(CONFIG_ARMV7_M_ARMV8_M_MAINLINE)
-	/* clear BASEPRI so wfi is awakened by incoming interrupts */
+#if defined(CONFIG_ARMV7_M_ARMV8_M_MAINLINE)
+	/*
+	 * PRIMASK is always cleared on ARMv7-M and ARMv8-M; set PRIMASK to
+	 * prevent interruption before wake-up.
+	 */
+	cpsid i
+
+	/*
+	 * Set wake-up interrupt priority to the lowest and synchronise to
+	 * ensure that this is visible for the WFI instruction execution.
+	 */
 	eors.n r0, r0
 	msr BASEPRI, r0
-#else
-#error Unknown ARM architecture
-#endif /* CONFIG_ARMV6_M_ARMV8_M_BASELINE */
+	isb
+
+#endif /* CONFIG_ARMV7_M_ARMV8_M_MAINLINE */
+
+	/* Ensure that all memory transactions are completed */
```

Comment:
Nope, that's DMB. the text for DSB should probably be "Ensure that no more instructions are executed before all memory transactions are completed", or something similar.

Reply:
well, "ensure ... is completed" would mean that it will block until "..." is completed, which has the same meaning as "no more instructions are executed before ..."

I will add a more verbose description, in case that is not clear enough.